compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/ConstantValueResolver.java

```diff
@@ -494,4 +510,53 @@ private void checkUniqueness(BLangConstant constant) {
             }
         }
     }
+
+    private void updateSymbolType(BLangConstant constant) {
+        BConstantSymbol symbol = constant.symbol;
+        if (symbol.type.getKind() != TypeKind.FINITE && symbol.value != null) {
+            BType singletonType = checkType(constant.expr, constant, symbol.value.value, symbol.type, symbol.pos);
+            if (singletonType != null) {
+                symbol.type = singletonType;
+            }
+        }
+    }
+
+    private BFiniteType createFiniteType(BLangConstant constant, BLangExpression expr) {
+        BTypeSymbol finiteTypeSymbol = Symbols.createTypeSymbol(SymTag.FINITE_TYPE,
+                Flags.asMask(EnumSet.noneOf(Flag.class)), Names.EMPTY, constant.symbol.pkgID, null,
+                constant.symbol.owner, constant.symbol.pos, VIRTUAL);
+        BFiniteType finiteType = new BFiniteType(finiteTypeSymbol);
+        finiteType.addValue(expr);
+        return finiteType;
+    }
+
+    private BType checkType(BLangExpression expr, BLangConstant constant, Object value, BType type, Location pos) {
+        if (expr != null && expr.getKind() == NodeKind.SIMPLE_VARIABLE_REF &&
+                ((BLangSimpleVarRef) expr).symbol.type.getKind() == TypeKind.FINITE) {
+            return ((BLangSimpleVarRef) expr).symbol.type; // Reference is already resolved.
+        }
+        switch (type.tag) {
+            case TypeTags.INT:
+            case TypeTags.BYTE:
```

Comment:
For byte constants (and other int subtypes when we support them), IMO we should create int finite types.

Reply:
Then, shall I remove that case?